### ISSUE TITLE
[Suggestion] New TSC defines system

### ISSUE THREAD

**Issue by Leo40Git** [taedixon/boosters-lab#20]
(Suggestion by GabrielRavier#5356 on Discord)
Have comments at the end of a script that store the defines, as such:
```
(normal events)
#LAST
<END
//#define FACE_SUE = 0001
//#define ARMS_BLADE = 0007```
**Pros:**
- More obvious system than the current one
- As long as ScriptSource is available, the defines are also available
**Cons:**
- Making global defines with this system would be difficult

**Comment by Leo40Git** [comment]
That could work.

**Comment by GabrielRavier** [comment]
A better idea (to avoid avoid to many defines and overrunning the TSC buffer) would be to just have a global define file so that there can be a bunch of default defines (musics, weapons, sound effects and all that)

**Comment by GabrielRavier** [comment]
Basically if we want this to be really useful then BL needs to automatically replace TSC args by their appropriate defined name (obviously you then need appropriate syntax for differenciating flags, events and others)

**Comment by Leo40Git** [comment]
Proposed 'strong-typed' `#define` format:
`#define <TYPE> <NAME> <VALUE>`
`<TYPE>`: Type of definition (using CE param types?)
Of course, that would make the "replace defines with their values" code more complicated than if defines were simply a name-value deal.

**Comment by taedixon** [comment]
doesn't this feature already exist

**Comment by taedixon** [comment]
I mean like, a shitty version of it that nobody uses, but still...

**Comment by Leo40Git** [comment]
Yes, that's why I called this suggestion "*New* TSC defines system".
